docs/terminals.md

```diff
@@ -0,0 +1,141 @@
+# Terminals and Standard IO
+All processes on Unix (and Unix-like) operating systems have 3 standard file descriptors (fd) passed to them at start, collectively standard-IO (`stdio`):
+
+* `0`: standard-in (`stdin`), the input stream into the process
+* `1`: standard-out (`stdout`), the output stream from the process
+* `2`: standard-error (`stderr`), the error stream from the process
+
+When creating and running a container via `runc`, it is important to take care to structure the stdio the new container's process receives.
+
+## Terminal Modes
+
+`runc` supports two distinct methods for passing stdio to the container's primary process:
+
+* pass-through
+* new terminal
+
+### Pass-Through
+In pass-through mode, the `stdio` passed to the `runc` call will be passed as is to the primary container. This means that if you do the following:
```

Comment:
This isn't totally accurate. In the case of non-detached, the `stdio` passed to the container are actually pipes (which are then read from by the still-running `runc` process). The reason this might be important to someone is so they know that `fd=2` in your example isn't *actually* the file `/tmp/log.out`.

Reply:
I am not sure I knew that, although it makes sense. Updating.